### PR TITLE
Bug 1914509: [network-operator][multus] Not redering with the DHCP daemonset if IPAM is in plugins list

### PR DESCRIPTION
**Pull request by bysnupy**
* Version: OCPv4.6+
* Description: 
   DHCP type in IPAM can also configure in plugins network configuration list, so it should provide the condition to render DHCP daemonset additionally.
* Reference: [[network-operator][multus] Not redering with the DHCP daemonset if IPAM is in plugins list](https://bugzilla.redhat.com/show_bug.cgi?id=1914509)

### CODE REVIEW

File: pkg/network/dhcp_daemon.go [hunk 1]
```diff
@@ -6,6 +6,28 @@ import (
 	"log"
 )
 
+// rawConfigHasIPAM determines if the rawConfig JSON data has IPAM element.
+func rawConfigHasIPAM(rawConfig map[string]interface{}) bool {
```
Comment (dougbtv) [inline pkg/network/dhcp_daemon.go:10]: Hey Daein, just a nit, but can we call this something related to the fact that it has DHCP in the IPAM, such as `rawConfigIPAMHasDHCP`? Thanks!

**Review by dougbtv — APPROVED** [review]
Dain, thanks for I'm personally OK with it as is. I like the changes overall, and thank you for filing the BZ and the solution :100: Just one comment regarding noting that it has DHCP in the IPAM for naming. Thanks!

File: pkg/network/dhcp_daemon.go (hunk 1 above)
Comment (bysnupy) [reply pkg/network/dhcp_daemon.go]: Good point, I've changed the function name as `rawConfigIPAMHasDHCP` as you mentioned. Thanks!

**Review by s1061123 — APPROVED** [review]
Looks good to me, thanks!

**Comment by bysnupy** [comment]
BZ assignee is changed from @fepan to @dougbtv , so I sync it.
/assign @dougbtv

**Comment by bysnupy** [comment]
@dougbtv I'm ready for merge. All the test is passed FINALLY.

**Comment by bysnupy** [comment]
@weliang1 Your approval did not apply on this PR. Do we need additional workflow for merging this PR ?

**Comment by bysnupy** [comment]
@dougbtv I'm sorry for bothering you. Could you know who can approve lgtm ? This ticket sees like floating.

**Comment by bysnupy** [comment]
Eventually, all tests passed.

**Pull request closed without merging**